app/src/main/java/memphis/myapplication/SendFriendRequest.java

```diff
@@ -59,23 +59,28 @@ protected void onCreate(Bundle savedInstanceState) {
         message = findViewById(R.id.trustTypeMessage);
 
         String friendsMessage = "";
-        friend = potentialFriends.get(0).getUsername();
-        User newFriend = realm.where(User.class).equalTo("username", friend).findFirst();
-        ArrayList<String> newFriendsList = newFriend.getFriends();
-        RealmResults<User> myFriends = realm.where(User.class).equalTo("trust", true).findAll();
-        ArrayList<String> myFriendsList = new ArrayList<>();
-        for (User u : myFriends) {
-            myFriendsList.add(u.getNamespace());
-        }
-        myFriendsList.retainAll(newFriendsList);
-        if (myFriendsList.isEmpty()) {
-            friendsMessage = "No mutual friends";
-        }
-        for (String f : myFriendsList) {
-            mutualFriend = f.substring(f.lastIndexOf("/")+1);
-            friendsMessage = friendsMessage + f + "\n";
+        if (potentialFriends.isEmpty()) {
+            friend = "";
+        } else {
+            friend = potentialFriends.get(0).getUsername();
+            User newFriend = realm.where(User.class).equalTo("username", friend).findFirst();
+            ArrayList<String> newFriendsList = newFriend.getFriends();
+            RealmResults<User> myFriends = realm.where(User.class).equalTo("trust", true).findAll();
+            ArrayList<String> myFriendsList = new ArrayList<>();
+            for (User u : myFriends) {
+                myFriendsList.add(u.getNamespace());
+            }
+            myFriendsList.retainAll(newFriendsList);
+            if (myFriendsList.isEmpty()) {
+                friendsMessage = "No mutual friends";
+            }
+            for (String f : myFriendsList) {
+                mutualFriend = f.substring(f.lastIndexOf("/")+1);
+                friendsMessage = friendsMessage + f + "\n";
+            }
+            message.setText(friendsMessage);
         }
-        message.setText(friendsMessage);
+
         realm.close();
```

Comment:
realm.close is also closed below? Something seems weird - is it the indentation?

Reply:
The second realm.close call should be in the onClick scope.